pkg/cmd/cli/cmd/deploy.go

```diff
@@ -222,7 +222,7 @@ func (o DeployOptions) deploy(config *deployapi.DeploymentConfig, out io.Writer)
 	}
 
 	config.Status.LatestVersion++
-	_, err = o.osClient.DeploymentConfigs(config.Namespace).Update(config)
+	_, err = o.osClient.DeploymentConfigs(config.Namespace).UpdateStatus(config)
```

Comment:
actually, we would never want a client to modify status to trigger a deployment... a client would modify spec, the controller would modify status to indicate the requested deployment was done